Simplified/Book/UI/NYPLBookCellDelegate+AudiobookProgressSaving.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  NYPLBookCellDelegate+AudiobookProgressSaving.swift
+//  Simplified
+//
+//  Created by Ernest Fan on 2022-03-01.
+//  Copyright © 2022 NYPL. All rights reserved.
+//
+#if FEATURE_AUDIOBOOKS
+import Foundation
+import NYPLAudiobookToolkit
+import UIKit
+
+private let NYPLAudiobookProgressSavingInterval: DispatchTimeInterval = .seconds(60)
+
+@objc extension NYPLBookCellDelegate {
+  
+  private func savePosition() {
+    NYPLBookRegistry.shared().save()
+  }
+  
+  // Create a timer that saves the audiobook progress periodically when the app is inactive.
+  // We do save the progress when app is being killed and applicationWillTerminate: is called,
+  // but applicationWillTerminate: is not always called when users force quit the app.
+  // This method is triggered when app resigns active.
+  @objc(scheduleProgressSavingTimerForAudiobookManager:)
+  func scheduleProgressSavingTimer(for manager: DefaultAudiobookManager?) {
+    guard let manager = manager else {
+      return
+    }
+
+    let audiobookProgressSavingQueue = DispatchQueue(label: "audiobookProgressSavingQueue")
+    
+    weak var weakManager = manager
+    
+    let timer = DispatchSource.repeatingTimer(interval: NYPLAudiobookProgressSavingInterval,
+                                              queue: audiobookProgressSavingQueue) { [weak self] in
```

Comment:
is the timer owning the queue?

Reply:
I look it up but couldn't find anything, the docs say `the dispatch queue to which to execute the installed handlers.`. Is there a specific reason why we don't want the timer to own the queue?

Reply:
the problem here is that if the timer does _not_ own the queue, the queue may be dealloced at any time, because we do not retain it here. I checked the docs too and couldn't find a clear answer. 

To be on the safe side we could have the `audiobookProgressSavingQueue` as a property of the CellDelegate, and initialize it externally, in the NYPLBookCellDelegate initializer for example. Having one queue should be fine, because after all we only listen to one book at a time. What do you think?

Reply:
I did not thought about the queue being dealloc, nice catch.
I think your suggestion would work, what exactly do you mean when you say initialize it **externally**?

Reply:
oh sorry I meant externally from this method. If you initialize it in the BookCellDelegate initializer it should work I think. It's just my first thought though...